### PR TITLE
Create GroupCache with builder

### PR DESCRIPTION
**Pull request by Pan-Maciek**
Summary:
- simplify Groupcache configuration during creation - using builder pattern
- remove a need for manual boxing of loader
- keep Options as internal structure - we don't want to pass multiple parameters around

```rust
// --- Case 1 ---
// Before
let cache = Groupcache::new(socket.into(), Box::new(loader));

// After
let cache = Groupcache::builder(socket.into(), loader).build();

// --- Case 2 ---
// Before
let options = OptionsBuilder::new()
    .hot_cache(hot_cache)
    .https()
    .build();

let cache = Groupcache:: new_with_options(socket.into(), Box::new(loader));

// After
let cache = Groupcache::builder(socket.into(), loader)
    .hot_cache(hot_cache)
    .https()
    .build();
```

### CODE REVIEW

File: groupcache/src/groupcache_builder.rs [hunk 1]
```diff
@@ -0,0 +1,82 @@
+use crate::groupcache::ValueBounds;
+use crate::options::Options;
+use crate::{Groupcache, GroupcacheInner, GroupcachePeer, ValueLoader};
+use moka::future::Cache;
+use std::sync::Arc;
+use tonic::transport::Endpoint;
+
+pub struct GroupcacheBuilder<Value: ValueBounds> {
```
Comment (Petroniuss) [inline groupcache/src/groupcache_builder.rs:9]: This is a public struct we may want to add documentation for it:
- ```Allows to customize groupcache: caches, timeouts etc```

File: groupcache/src/lib.rs [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 mod errors;
 mod groupcache;
+pub mod groupcache_builder;
```
Comment (Petroniuss) [inline groupcache/src/lib.rs]: I wouldn't expose it under a different mod: It should be exported under ::groupcache - there's no need to have multiple modules externally for a rather simple crate

**Comment by Petroniuss** [comment]
I like it <3  Makes the API more fluent.

File: groupcache/src/lib.rs (hunk 2 above)
Comment (Pan-Maciek) [reply groupcache/src/lib.rs]: I'm pretty sure it would get exposed under the same ::groupcache name, but exposing it more explicitly with pub use

**Review by Petroniuss — APPROVED** [review]
Approved :)

**Pull request merged**